xcp_d/utils/utils.py

```diff
@@ -119,125 +119,130 @@ def get_transformfilex(bold_file, mni_to_t1w, t1w_to_native):
-                               + '*_mode-image_xfm.h5')[0]
-        transformfileMNI = [str(pnc_to_t1w), str(t1w_to_mni)]
-        transformfileT1W = str(pnc_to_t1w)
-
-    elif 'space-NKI' in file_base:
-        mnisf = mni_to_t1w.split('from-')[0]
-        nki_to_t1w = mnisf + 'from-NKI_to-T1w_mode-image_xfm.h5'
-        t1w_to_mni = glob.glob(mnisf + 'from-T1w_to-' + template
-                               + '*_mode-image_xfm.h5')[0]
-        transformfileMNI = [str(nki_to_t1w), str(t1w_to_mni)]
-        transformfileT1W = str(nki_to_t1w)
-
-    elif 'space-OASIS' in file_base:
-        mnisf = mni_to_t1w.split('from')[0]
-        oasis_to_t1w = mnisf + 'from-OASIS30ANTs_to-T1w_mode-image_xfm.h5'
-        t1w_to_mni = glob.glob(mnisf + 'from-T1w_to-' + template
-                               + '*_mode-image_xfm.h5')[0]
-        transformfileMNI = [str(oasis_to_t1w), str(t1w_to_mni)]
-        transformfileT1W = [str(oasis_to_t1w)]
-
-    elif 'space-MNI152NLin6Sym' in file_base:
-        mnisf = mni_to_t1w.split('from-')[0]
-        mni6c_to_t1w = mnisf + 'from-MNI152NLin6Sym_to-T1w_mode-image_xfm.h5'
-        t1w_to_mni = glob.glob(mnisf + 'from-T1w_to-' + template
-                               + '*_mode-image_xfm.h5')[0]
-        transformfileMNI = [str(mni6c_to_t1w), str(t1w_to_mni)]
-        transformfileT1W = [str(mni6c_to_t1w)]
-
-    elif 'space-MNIInfant' in file_base:
-        transformfileMNI = str(
-            pkgrf('xcp_d', 'data/transform/infant_to_2009_Composite.h5'))
-        transformfileT1W = str(mni_to_t1w)
-
-    elif 'space-MNIPediatricAsym' in file_base:
-        mnisf = mni_to_t1w.split('from-')[0]
-        mni6c_to_t1w = glob.glob(
-            mnisf + 'from-MNIPediatricAsym*_to-T1w_mode-image_xfm.h5')[0]
-        t1w_to_mni = glob.glob(mnisf + 'from-T1w_to-' + template
-                               + '*_mode-image_xfm.h5')[0]
-        transformfileMNI = [str(mni6c_to_t1w), str(t1w_to_mni)]
-        transformfileT1W = [str(mni6c_to_t1w)]
-
-    elif 'space-T1w' in file_base:
-        mnisf = mni_to_t1w.split('from')[0]
-        transformfileMNI = [str(t1w_to_mni)]
-        transformfileT1W = [
-            str(pkgrf('xcp_d', 'data/transform/oneratiotransform.txt'))
-        ]
+    # Extract the space of the BOLD file
+    file_base = os.path.basename(bold_file)
+    bold_space = re.findall("space-([a-zA-Z0-9]+)", file_base)
+    if not len(bold_space):
+        bold_space = "native"
+    else:
+        bold_space = bold_space[0]
+
+    if bold_space in ("native", "T1w"):
+        base_std_space = re.findall("from-([a-zA-Z0-9]+)", mni_to_t1w)[0]
+    elif f"from-{bold_space}" not in mni_to_t1w:
+        raise ValueError(f"Transform does not match BOLD space: {bold_space} != {mni_to_t1w}")
+
+    # Pull out the correct transforms based on bold_file name and string them together.
+    xforms_to_T1w = [mni_to_t1w]  # used for all spaces except T1w and native
```

Comment:
I don't think `mni_to_t1w` is guaranteed to be from MNI152NLin2009cAsym. It can be MNI152NLin2009cAsym, MNI152NLin6Asym, or MNIInfant, at the moment.

`xforms_to_MNI` could be renamed `xforms_to_ MNI152NLin2009cAsym` though.